### DISCUSSION TITLE
Search only properly works with ransack with a paid license

### DISCUSSION THREAD

**Discussion by djalmaaraujo** [avo-hq/avo#2296] [Feature feedback]
I tried for many hours to make `query.ransack` to work with a non-licensed version of avo. The resource search bar is displayed, but the console shows NO query at all with ransack.

I imagine there's a incompatibility with the community version of avo, but I don't see any mention in the docs: https://docs.avohq.io/3.0/search.html#search.

I can see that `description` only works with the pro version, but I would guess simply search would work. Once I applied the license (advanced) everything just started to work.

I am not sure how to report as a bug, because my guess is this is a bad-documentation rather than a bug.

Can you confirm my suspicious?

Thanks.

Example code:

```ruby
self.search = {
  query: -> {
    query.ransack(pid_or_reference_id_or_destination_value_or_content_cont: params[:q], m: "or").result(distinct: true)
  },
  item: -> do
    {
      title: "#{record.status} — #{record.content.truncate(60)}…",
    }
  end
}
```

This code was not working. The rails server was showing "Select from messages" without any condition at all. Once I applied the license, everything just works.

**Comment by adrianthedev** [comment]
Hmm. That should not happen.

I tested it with these settings and it works fine.

Are you using `pundit`?
Is it a new or existing app?

### Reproduction

```bash
rails new search_test
cd search_test
bin/rails app:template LOCATION='https://avohq.io/app-template'
bin/rails g model user name:string
rails db:migrate
bundle add ransack
rails s
```

Also add this to `app/models/application_record.rb`

```ruby
  # `ransackable_attributes` by default returns all column names
  # and any defined ransackers as an array of strings.
  # For overriding with a whitelist array of strings.
  #
  def self.ransackable_attributes(auth_object = nil)
    authorizable_ransackable_attributes
  end

  # `ransackable_associations` by default returns the names
  # of all associations as an array of strings.
  # For overriding with a whitelist array of strings.
  #
  def self.ransackable_associations(auth_object = nil)
    authorizable_ransackable_associations
  end
```
